### PR TITLE
add dngrep.json

### LINKED ISSUE #7712: qtranslate download url error
## Bug Report

**Package Name:** qtranslate

### Current Behaviour
 download failed

### Expected Behaviour
download successful

### Additional context/output
qtranslate.json download address is https://qtranslate2.appspot.com/QTranslate.6.9.0.zip

### Possible Solution
The current download address is http://qtranslate1.appspot.com/QTranslate.6.9.0.exe

### System details

**Windows version:** [10]

**OS architecture:** [64bit]

### LINKED ISSUE #7719: add dnGREP
### Information

**Name:** dnGREP

**Description:** File and content search-and-replace program for windows

**Homepage:** http://dngrep.github.io/

**Download link(s):** https://github.com/dnGrep/dnGrep/releases/

**Some indication of popularity/repute:** [Github stars 479](http://dngrep.github.io/)

Issue before PR.

### PR DESCRIPTION
**Pull request by RCJacH**
I apologize for the last PR, had trouble with the old repository, deleted and re-forked.

Tested installation, uninstallation, and autoupdate.
The file `dnGREP.Settings.dat` is generated only after opening settings menu for the first time, and adding it to Persist will create a new empty folder with the same name, thus ignored.

Closes #7719

Relates to #XXXX

- [ ] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/dngrep.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "version": "2.9.482",
+    "description": "Powerful search for Windows",
```
Comment (rashil2000) [inline bucket/dngrep.json:3]:
```suggestion
    "description": "Graphical GREP tool to search across text files, Word and Excel documents, PDFs and archives",
```

File: bucket/dngrep.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            "url": "https://github.com/dnGrep/dnGrep/releases/download/v2.9.482.0/dnGREP.2.9.482.x64.msi",
+            "hash": "59cadfa678eca183ebdd9840455da3b3ac9872eb69b39d8175bc6594ebe0532f"
+        },
+        "32bit": {
+            "url": "https://github.com/dnGrep/dnGrep/releases/download/v2.9.482.0/dnGREP.2.9.482.x86.msi",
+            "hash": "ffe7136771bd90fb884587b89341a64a2fd2b0cc4008e55222cd496e47a9d813"
+        }
+    },
+    "extract_dir": "dnGREP",
+    "bin": "dnGREP.exe",
```
Comment (rashil2000) [inline bucket/dngrep.json:17]: Any commandline arguments?

File: bucket/dngrep.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "shortcuts": [
+        [
+            "dnGREP.exe",
+            "dnGREP"
+        ]
+    ],
+    "persist": [
+        "Themes"
+    ],
```
Comment (rashil2000) [inline bucket/dngrep.json:26]:
```suggestion
    "persist": "Themes",
```

File: bucket/dngrep.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    "checkver": {
+        "github": "https://github.com/dnGrep/dnGrep",
+        "regex": "Name: dnGREP.([\\d.]+).x64"
```
Comment (rashil2000) [inline bucket/dngrep.json:29]:
```suggestion
        "github": "https://github.com/dnGrep/dnGrep"
```

Regex not required

**Review by rashil2000 — CHANGES_REQUESTED** [review]
Add hash in autoupdate, from here:

image

File: bucket/dngrep.json (hunk 2 above)
Comment (RCJacH) [reply bucket/dngrep.json:17]: the commandline arguments are simply settings to open dnGREP gui with.
What should I do with them?

Comment (rashil2000) [reply bucket/dngrep.json:17]: Can you show a link to them (any docs/screenshots etc.)

Comment (RCJacH) [reply bucket/dngrep.json:17]: https://github.com/dnGrep/dnGrep/wiki/Command-Line

Comment (rashil2000) [reply bucket/dngrep.json:17]: Okay, nothing to change here in that case

File: bucket/dngrep.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        "64bit": {
+            "url": "https://github.com/dnGrep/dnGrep/releases/download/v2.9.482.0/dnGREP.2.9.482.x64.msi",
+            "hash": "59cadfa678eca183ebdd9840455da3b3ac9872eb69b39d8175bc6594ebe0532f"
+        },
+        "32bit": {
+            "url": "https://github.com/dnGrep/dnGrep/releases/download/v2.9.482.0/dnGREP.2.9.482.x86.msi",
+            "hash": "ffe7136771bd90fb884587b89341a64a2fd2b0cc4008e55222cd496e47a9d813"
+        }
+    },
+    "pre_install": "if (!(Test-Path \"$persist_dir\\dnGREP.Settings.dat\")) { New-Item \"$dir\\dnGREP.Settings.dat\" | Out-Null }",
```
Comment (rashil2000) [inline bucket/dngrep.json:16]: Aren't .dat files binary formatted? Can you confirm if dnGrep is able to read/write to this file when it's created by Scoop here?

Comment (RCJacH) [reply bucket/dngrep.json:16]: it's actually a xml file.
I can confirm that after creating an empty file with this name and moving to persist folder, when opening Options windows in dnGREP gui, it populates this file with default values.

Comment (rashil2000) [reply bucket/dngrep.json:16]: Okay, thanks for the info!

**Pull request merged**